packages/graphql_flutter/example/graphql_chat/lib/view/home/home_body.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
+import 'package:graphql_chat/api/query.dart';
+import 'package:graphql_chat/model/chat.dart';
+import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:logger/logger.dart';
+
+class HomeViewBody extends StatelessWidget {
+  final Logger _logger = Logger();
+  List<Chat> _chats = [];
+
+  HomeViewBody({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const SizedBox(width: 20, height: 90),
+        Expanded(child: _buildBodyView(context: context))
+      ],
+    );
+  }
+
+  Widget _buildScrollView({required BuildContext context}) {
+    return Query(
+        options: QueryOptions<List<Chat>>(
+            fetchPolicy: FetchPolicy.networkOnly,
+            parserFn: (Map<String, dynamic> json) {
+              var rawList = List.of(json["getChats"]);
+              return rawList
+                  .map((jsonChat) => Chat.fromJSON(jsonChat))
+                  .toList();
+            },
+            document: gql(Queries.getGetQuery())),
+        builder: (QueryResult result,
+            {VoidCallback? refetch, FetchMore? fetchMore}) {
+          if (result.hasException) {
+            _logger.e(result.exception);
+            throw Exception(result.exception);
+          }
+          if (result.isLoading) {
+            _logger.i("Still loading");
+            return const Text("Loading chats");
+          }
+          _logger.d(result.data ?? "Data is undefined");
+          _chats = result.parsedData as List<Chat>;
```

Comment:
Instead of storing the data on build, we should move the query up into the main `build` method and pass the data down to the children.

Reply:
I see, I will try to refactoring the code with this suggestion, thanks!